packager/VisibilityChecker.cc

```diff
@@ -468,8 +432,34 @@ class VisibilityCheckerPass final {
                               path.has_value();
             }
             if (!causesCycle && !layeringViolation && !strictDependenciesTooLow) {
-                if (wasNotImported) {
-                    // We failed to import the package that defines the symbol
+                if (!isExported) {
```

Comment:
Here, `!isExported` and `!wasImported` are treated as exclusive, but it's actually possible for both to be true. In a future PR, I'll handle the `!isExported` && !wasImported` case.